src/training/ui/mod.rs

```diff
@@ -33,10 +34,11 @@ pub unsafe fn handle_draw(layout: *mut Layout, draw_info: u64, cmd_buffer: u64)
     original!()(layout, draw_info, cmd_buffer);
 }
 
-// We'll keep some sane max size here; we shouldn't reach above 600KiB is the idea,
-// but we can try higher if we need to.
+// Allocate a static amount of memory that Smash isn't allowed to deallocate,
+// in order for us to be able to swap the 'layout.arc' with the current
+// version of the file in between loads of training mode.
 #[cfg(feature = "layout_arc_from_file")]
-static mut LAYOUT_ARC: &mut [u8; 600000] = &mut [0u8; 600000];
+static mut LAYOUT_ARC: &mut [u8; (2 * MEBIBYTE) as usize] = &mut [0u8; (2 * MEBIBYTE) as usize];
```

Comment:
Love this. You could also add it as a separate constant if you want.
```rust
#[cfg(feature = "layout_arc_from_file")]
const LAYOUT_ARC_SIZE : usize = (2 * MEBIBYTE) as usize;
#[cfg(feature = "layout_arc_from_file")]
static mut LAYOUT_ARC: &mut [u8; LAYOUT_ARC_SIZE] = &mut [0u8; LAYOUT_ARC_SIZE];
```

Reply:
Great suggestion, definitely adding this

Reply:
Incorporated in https://github.com/jugeeya/UltimateTrainingModpack/pull/482/commits/c7057f301387b84377942d49d15e498a3afcdb29